### PR TITLE
Feature/upgrade django version

### PR DESCRIPTION
**Pull request by xtornasol512**
# DESCRIPTION

Upgrade to newer django version 4.2 

## Changes

- Redis upgrade
- Database Index
- New packages
- New python version supported

Extras:
- fix some flake8 warnings
- Use of django-upgrade on "urls" files i.e.:  `$ django-upgrade --target-version 4.2 api/urls.py`

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,17 +1,17 @@
 # django
-Django==3.2.18
+Django==4.2.16
```
Comment (ebarojas) [inline requirements.txt:2]: 🔥

**Review by ebarojas — APPROVED** [review]
👍 

🚀

**Comment by JasonBerto** [comment]
Qa done, 

Jesus, puedes actualizar el tag, no tengo permisos en este repo

**Review by ebarojas — APPROVED** [review]
🚀

**Pull request merged**